src/lavinmq/queue/message_store.cr

```diff
@@ -383,6 +385,11 @@ module LavinMQ
         end
       end
 
+      private def log_progress_and_yield(message)
+        Log.info { message }
```

Comment:
One "problem" here is that if you set log level to something higher than `info` the message string will still be created for every call to `log_progress_and_yield`. It's probably better to do 
```suggestion
      private def log_progress_and_yield(&block)
        Log.info &block
```
and change all calls to `log_progress_and_yield` to use a block: `log_progress_and_yield { "hello world" }`

Reply:
I'm not getting any logs when using this approach. Any ideas?

``` 
private def log_progress_and_yield(&block)
  Log.info &block
  Fiber.yield
end
```

called by for example:
```
log_progress_and_yield { "Deleting unused segment #{seg}" }
```

Reply:
Solved by using `DirectDispatcher` instead, and just logging like normal in this file